Caleydo/ordino_public
# Redesign welcome page and start menu

Requires PR https://github.com/Caleydo/ordino/pull/260

Part of Caleydo/tdp_bi_bioinfodb#1096

### Summary

* Introduce the new welcome page and start menu
   * Start menu is implemented and discussed in https://github.com/Caleydo/ordino/pull/260/
   * Welcome page is currently implemented and discussed here
* Use React, React Bootstraps, and Bootstrap 4 for new components
   * There are currently some issues due to BS4 only (see https://github.com/datavisyn/visyn2020/issues/10#issuecomment-752918269 for further information)

### How to test

1. Setup an Ordino Public workspace (`yo phovea:setup-workspace Caleydo/ordino_product -b welcome-page`) or switch your branches accordingly
2. Copy the [welcome.scss](https://github.com/Caleydo/ordino_product/blob/welcome-page/templates/web/welcome.scss) to the workspace root (if not already available)
2. `npm start` and `docker-compose up`
3. Depending on the configured entry points in ordino_public you should either see the welcome page or the Ordino app

### Known Issues

* Bad performance of the welcome page, as the whole Phovea, TDP, and node_modules are bundled also for the welcome page (which would only require react and react-router). Workaround to create smaller welcome page bundle:
   1. [Use welcome entry point in the .yo-rc.json](https://github.com/Caleydo/ordino_public/commit/0fafe491bdeeafa16cf9e5d38629dc8ce8b4e236) by renaming them
   2. Avoid bundling the phovea_registry in the webpack.dev.js:
      ```diff
      - entry: webpackHelper.injectRegistry(workspacePath, defaultAppPath, [workspaceRegistryFile], entries),
      + entry: webpackHelper.injectRegistry(workspacePath, defaultAppPath, [], entries),
      ```

### To dos

- [ ] Implement remaining pages in https://github.com/Caleydo/ordino_public/tree/welcome-page/src/pages according to click prototype
- [ ] Assess whether we can move some React components of the welcome page from ordino_public to this repository (and encourage reusability over copy+paste to other customizations)
- [ ] Improve Webpack bundles with tree shaking and code splitting to improve performance of welcome page
